src/qtism/data/storage/xml/marshalling/TemplateElementMarshaller.php

```diff
@@ -56,14 +60,21 @@ protected function unmarshallChildrenKnown(DOMElement $element, QtiComponentColl
                         $component->setShowHide(ShowHide::getConstantByName($showHide));
                     } catch (InvalidArgumentException $e) {
                         $msg = "'${showHide}' is not a valid value for the 'showHide' attribute of element '" . $element->localName . "'.";
+
                         throw new UnmarshallingException($msg, $element, $e);
                     }
 
                     try {
-                        $content = ($element->localName === 'templateInline') ? new InlineStaticCollection() : new FlowStaticCollection();
+                        $content = ($element->localName === 'templateInline') ? new InlineStaticCollection(
+                        ) : new FlowStaticCollection();
                         foreach ($children as $c) {
-                            if (!$c instanceof FlowStatic || ($element->localName === 'templateBlock' && in_array($c->getQtiClassName(), ['hottext', 'rubricBlock', 'infoControl']))) {
-                                $msg = "The '" . $element->localName . "' cannot contain '" . $c->getQtiClassName() . "' elements.";
+                            if (!$c instanceof FlowStatic || ($element->localName === 'templateBlock' && in_array(
+                                        $c->getQtiClassName(),
+                                        ['hottext', 'rubricBlock', 'infoControl']
+                                    ))) {
```

Comment:
could you provide here proper multiline if statement as suggested earlier or revert that place?